src/Libraries/GeometryUI/ExportWithUnits.cs

```diff
@@ -1,17 +1,103 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Xml;
-using Autodesk.DesignScript.Geometry;
+﻿using Autodesk.DesignScript.Geometry;
 using Dynamo.Graph;
 using Dynamo.Graph.Nodes;
 using Dynamo.Utilities;
 using DynamoConversions;
 using GeometryUI.Properties;
 using Newtonsoft.Json;
 using ProtoCore.AST.AssociativeAST;
+using System;
+using System.Collections.Generic;
+using System.Xml;
 
 namespace GeometryUI
 {
+
+    //TODO move to own importwithunits file.
+    //TODO finish the descriptions etc with resx.
+
+    [NodeCategory(BuiltinNodeCategories.GEOMETRY)]
+    [NodeName("Geometry.ImportFromSatAndUnits")]
+    [InPortTypes(new string[] { "var", "DynamoUnits.Unit" })]
+    //[NodeDescription("ExportToSATDescripiton", typeof(GeometryUI.Properties.Resources))]
+    //[NodeSearchTags("ExportWithUnitsSearchTags", typeof(GeometryUI.Properties.Resources))]
+    [OutPortTypes("Geometry[]")]
+    [IsDesignScriptCompatible]
+    public class ImportFromSATAndUnits : NodeModel
+    {
+        [JsonConstructor]
+        private ImportFromSATAndUnits(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts) :
+    base(inPorts, outPorts)
+        {
+            ShouldDisplayPreviewCore = true;
+        }
+
+        public ImportFromSATAndUnits()
+        {
+            InPorts.Add(new PortModel(PortType.Input, this, new PortData("file|filePath", "TODO")));
+            InPorts.Add(new PortModel(PortType.Input, this, new PortData("dynamoUnit", "TODO", new NullNode())));
+            OutPorts.Add(new PortModel(PortType.Output, this, new PortData("geometry", "TODO")));
+
+            ShouldDisplayPreviewCore = true;
+            RegisterAllPorts();
+        }
+
+        public override IEnumerable<AssociativeNode> BuildOutputAst(
+          List<AssociativeNode> inputAstNodes)
+        {
+            if (!InPorts[0].IsConnected || (!InPorts[1].IsConnected && !InPorts[1].UsingDefaultValue))
+            {
+                return new[] { AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(0), AstFactory.BuildNullNode()) };
+            }
+
+            var funcNode = AstFactory.BuildFunctionCall("ImportHelpers", "ImportFromSATByUnits",
+                        new List<AssociativeNode> { inputAstNodes[0], inputAstNodes[1] });
+
+            return new[] { AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(0), funcNode) };
+        }
```

Comment:
What about creating a function object and all that boilerplate code if the node is not fully satisfied?

Reply:
And also lacing?

Reply:
I was following the example from the export node which does not handle any of this, because this node doesn't have any UI I think it's more reasonable to expect that boiler plate behavior, will add.